exporter/datadogexporter/factory.go

```diff
@@ -501,7 +514,28 @@ func (f *factory) createLogsExporter(
 		return nil, fmt.Errorf("failed to build attributes translator: %w", err)
 	}
 
-	if cfg.OnlyMetadata {
+	if isLogsAgentExporterEnabled() {
+		logComponent, _ := newLogComponent(set.TelemetrySettings)
+		cfgComponent, _ := newConfigComponent(set.TelemetrySettings, cfg)
+		hostname := logs.NewHostnameService(pcfg.ConfigHostname)
```

Comment:
Not sure if this is the correct hostname to use in logs, @mx-psi would you be able to take a look?

Reply:
I don't have enough context here. What does a hostname service do? It seems like this is wrong in that we are not using the full fallback hostname mechanism, but I need more details here

Reply:
The hostname service is mainly an abstraction of hostname.Get from datadog-agent, this is what it looks like in the datadog-agent repository: https://github.com/DataDog/datadog-agent/blob/main/comp/core/hostname/hostnameimpl/service.go.

How would we modify the implementation to use the full fallback hostname mechanism?

Reply:
You have to get the source provider, you can use this function https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/247a9f996e09a83cdc25addf70c05e42b8b30186/exporter/datadogexporter/factory.go#L99

Reply:
There is one more change to make, see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32327#discussion_r1587459860